src/main/java/hudson/plugins/build_timeout/impl/ElasticTimeOutStrategy.java

```diff
@@ -68,9 +81,17 @@ public long getTimeOut(AbstractBuild<?, ?> build, BuildListener listener)
             throws InterruptedException, MacroEvaluationException, IOException {
         double elasticTimeout = getElasticTimeout(Integer.parseInt(expandAll(build,listener,getTimeoutPercentage())), build, listener);
         if (elasticTimeout == 0) {
-            return Math.max(MINIMUM_TIMEOUT_MILLISECONDS, Integer.parseInt(expandAll(build, listener, getTimeoutMinutesElasticDefault())) * MINUTES);
+            if (isFailSafeTimeoutDuration()) {
+                return Math.max(Integer.parseInt(getTimeoutMinutesElasticDefault()) * 60L * 1000L, Integer.parseInt(expandAll(build, listener, getTimeoutMinutesElasticDefault())) * MINUTES);
```

Comment:
This results `timeoutMinutesElasticDefault` anyway.
I don't think we need this `if` switch.